packages/core/src/types.ts

```diff
@@ -4,6 +4,32 @@ export type StringTransformer = (generated: string, descriptor: string) => strin
 
 export type ObjectTransformer = (generated: CSSProperties, descriptor: CSSProperties) => CSSProperties
 
+export type CCSSInput = any
+
+export type CCSSOriginal = any
+
+export type CCSSAnyFunction = (...args: any[]) => any
+
+export type CCSSToCSSRule = (
+    cssProp: string,
+    objectProp: string,
+    input: CCSSInput,
+    prop: string,
+    options: CCSSOptions
+) => CSSProperties | string
+
+export type CCSSToPseudoChild = (input: CCSSInput, prop: string, options: CCSSOptions) => CSSProperties | CCSSInput
+
+export interface IOutputTransformer {
+    (generated: string, descriptor: string): string
+    (generated: CSSProperties, descriptor: CSSProperties): CSSProperties
+    toCSSRule: CCSSToCSSRule
+    toPseudo: CCSSToPseudoChild
+    defaultOutput: CCSSAnyFunction
+    toChild: CCSSToPseudoChild
+    unsupportedHandler: (generated: any, input: any, prop: any) => Object | string
+}
```

Comment:
Small detail here:
This is how we should write our `Function Types`. The first two likes of this interface will have two constructors. The subsequent lines are just properties.

Reply:
You need overrides then.

Reply:
The two first lines of the interface actually are the overrides :D